homeassistant/helpers/entity.py

```diff
@@ -767,6 +773,7 @@ def add_to_platform_start(
         self.platform = platform
         self.parallel_updates = parallel_updates
         self._added = True
+        self._entity_removed = False
```

Comment:
I was concerned that `add_to_platform_abort` set `self._added = False` and it might have an unexpected side effect if the entity was added later after that was called.  If thats not an issue, I think we can go with checking `_added`

Reply:
Just tried that and it also appears to break `update_before_add`

Reply:
We need a trinary state for this

Reply:
I made it an enum instead.